### PR TITLE
Migrate tests to null-safety

### PR DESCRIPTION
**Pull request by goderbauer**

### CODE REVIEW

File: packages/flutter/test/material/time_picker_test.dart [hunk 1]
```diff
@@ -55,7 +53,7 @@ class _TimePickerLauncher extends StatelessWidget {
   }
 }
 
-Future<Offset> startPicker(
+Future<Offset?> startPicker(
```
Comment (justinmc) [inline packages/flutter/test/material/time_picker_test.dart:56]: First time I've seen nullable type parameters.

File: packages/flutter/test/material/time_picker_test.dart [hunk 2]
```diff
@@ -419,7 +417,7 @@ void _tests() {
     );
 
     // Ensure we preserve day period as we roll over.
-    final dynamic pickerState = tester.state(_timePickerDialog);
+    final dynamic pickerState = tester.state(_timePickerDialog); // ignore: unnecessary_nullable_for_final_variable_declarations
```
Comment (justinmc) [inline packages/flutter/test/material/time_picker_test.dart:420]: Why was the ignore needed here? For my own curiosity.

File: packages/flutter/test/material/toggle_buttons_test.dart [hunk 3]
```diff
@@ -170,50 +168,6 @@ void main() {
     },
   );
 
-  testWidgets('children property cannot be null', (WidgetTester tester) async {
```
Comment (justinmc) [inline packages/flutter/test/material/toggle_buttons_test.dart:173]: Awesome that we can remove whole tests.

**Review by justinmc — APPROVED** [review]
LGTM 👍

File: packages/flutter/test/material/toggle_buttons_test.dart (hunk 3 above)
Comment (shihaohong) [inline packages/flutter/test/material/toggle_buttons_test.dart:173]: Pretty cool side effect of NNBD

**Review by shihaohong — APPROVED** [review]
LGTM

File: packages/flutter/test/material/time_picker_test.dart (hunk 2 above)
Comment (shihaohong) [reply packages/flutter/test/material/time_picker_test.dart:420]: I'm curious too, tried looking in the codebase for some explanation but couldn't find one. It seems like it has something to do with using `dynamic` as a type?

Comment (goderbauer) [reply packages/flutter/test/material/time_picker_test.dart:420]: NNBD comes with this new lint: https://dart-lang.github.io/linter/lints/unnecessary_nullable_for_final_variable_declarations.html

Basically, if you have a final variable and assign a non-null value to it, the type of that variable should also be non-null. This makes totally sense in normal code. In this case however, we do not have access to the actual type `_TimePickerDialogState`, which is a private not visible in the test. To still get access to it, the test declares it as dynamic, so we can call any method we want on it and test the internals of the private class. It's a hack, you should never to this in application code.

Unfortunately, dynamic is a nullable type, triggering the warning.

Comment (goderbauer) [reply packages/flutter/test/material/time_picker_test.dart:420]: I've actually filed https://github.com/dart-lang/linter/issues/2282 with the linter team to see if this lint is appropriate for `dynamic` variables. It does seem a little odd.

**Pull request merged**